File: PYTHON/motorcycle.py
from vehicle import Vehicle

class Motorcycle(Vehicle):
    def __init__(self, plat_nomor="", merk="", tahun_produksi="", warna="", jenis_motor="", kapasitas_tangki=""):
        super().__init__(plat_nomor, merk, tahun_produksi, warna)
        self._jenis_motor = jenis_motor
        self._kapasitas_tangki = kapasitas_tangki

    @property
    def jenis_motor(self):
        return self._jenis_motor

    @property
    def kapasitas_tangki(self):
        return self._kapasitas_tangki

    def display_info(self):
        return f"{super().display_info()}, Jenis Motor: {self.jenis_motor}, Kapasitas Tangki: {self.kapasitas_tangki}"

File: CPP/motorcycle.cpp
#pragma once
#include <string>
#include <vector>
#include <iostream>
#include "vehicle.cpp"

class Motorcycle : public Vehicle {
private:
    std::string _jenis_motor;
    std::string _kapasitas_tangki;

public:
    Motorcycle(std::string plat_nomor = "", std::string merk = "", int tahun_produksi = 0, std::string warna = "", std::string jenis_motor = "", std::string kapasitas_tangki = "");
    
    std::string getJenisMotor();
    std::string getKapasitasTangki();
    
    std::string displayInfo();
};

Motorcycle::Motorcycle(std::string plat_nomor, std::string merk, int tahun_produksi, std::string warna, std::string jenis_motor, std::string kapasitas_tangki)
    : Vehicle(plat_nomor, merk, tahun_produksi, warna), _jenis_motor(jenis_motor), _kapasitas_tangki(kapasitas_tangki) {}

std::string Motorcycle::getJenisMotor() {
    return _jenis_motor;
}

std::string Motorcycle::getKapasitasTangki() {
    return _kapasitas_tangki;
}

std::string Motorcycle::displayInfo() {
    return Vehicle::displayInfo() + ", Jenis Motor: " + _jenis_motor + ", Kapasitas Tangki: " + _kapasitas_tangki;
}
